.buildkite/scripts/spotless.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+if [[ -z "${BUILDKITE_PULL_REQUEST:-}" ]]; then
+  echo "Not a pull request, skipping spotless"
+  exit 0
+fi
+
+if ! git diff --exit-code; then
+  echo "Changes are present before running spotless, not running"
+  git status
+  exit 0
+fi
+
+NEW_COMMIT_MESSAGE="[CI] Auto commit changes from spotless"
+PREVIOUS_COMMIT_MESSAGE="$(git log -1 --pretty=%B)"
+
+echo "--- Running spotless"
+.ci/scripts/run-gradle.sh spotlessApply
```

Comment:
Let's add `-Dscan.tag.NESTED` here so we categorize this as a "nested" build. This can be helpful in filtering out these kinds of "inner" CI build executions in Gradle Enterprise.